rdflib/plugins/sparql/parser.py

```diff
@@ -59,7 +59,7 @@ def expandTriples(terms):
             elif isinstance(t, list):
                 # BlankNodePropertyList
                 # is this bnode the object of previous triples?
-                if (i % 3) == 2:
```

Comment:
Well done for debugging and finding this! I *wrote* this code, and I had no idea how it worked any more :)